src/mem.rs

```diff
@@ -0,0 +1,60 @@
+// NOTE Copied verbatim from the rlibc crate
```

Comment:
Ok, I just really feel like it's a bad idea to copy-paste code like this, especially as we know there are bugs that will now need to get fixed in multiple locations. It seems like there should be some way to use rlibc without forcing a crates.io crate, presumably similar to whatever will happen to *this* crate.